libs/databrickscfg/loader.go

```diff
@@ -76,6 +76,9 @@ func (l profileFromHostLoader) Configure(cfg *config.Config) error {
 	// Normalized version of the configured host.
 	host := normalizeHost(cfg.Host)
 	match, err := findMatchingProfile(configFile, func(s *ini.Section) bool {
+		if cfg.Profile != "" {
+			return cfg.Profile == s.Name()
+		}
```

Comment:
This should not be done here because the loader here only cares about the host in the profile. Matching on the profile is done by the SDK and if it finds a match we'd never get to this loader.